mesonbuild/dependencies/base.py

```diff
@@ -46,6 +46,25 @@ class DependencyException(MesonException):
     '''Exceptions raised while trying to find dependencies'''
 
 
+class MissingCompiler:
+    """Represent a None Compiler - when no tool chain is found.
+    replacing AttributeError with DependencyException"""
+
+    exception_msg = 'no toolchain found'
+
+    def __getattribute__(self, item):
+        raise DependencyException(MissingCompiler.exception_msg)
```

Comment:
Is there any particular reason you choose to remove `__getattribute__` (first) over `__getattr__` (last)?

Reply:
I don't think it makes a big difference. The main reason I suggested to remove the one I did was because that would allow using `self.exception_msg`, and only after leaving that comment and continuing on in my review did I realize that it could be simplified further to inline the message.

Reply:
I don't think it makes a big difference. The main reason I suggested to remove the one I did was because that would allow using `self.exception_msg`, and only after leaving that comment and continuing on in my review did I realize that it could be simplified further to inline the message.